olive/evaluator/olive_evaluator.py

```diff
@@ -590,16 +622,32 @@ class OliveEvaluatorConfig(ConfigBase):
     @validator("metrics")
     def validate_metrics(cls, v):
         metric_len = len(v)
-        if metric_len == 1:
-            return v
 
         metric_names = set([metric.name for metric in v])
         assert len(metric_names) == metric_len, "Metric names must be unique"
 
-        rank_set = set([metric.priority_rank for metric in v])
-        expected_rank_set = set(range(1, metric_len + 1))
+        sub_type_names = set()
+        sub_type_with_rank = set()
+        rank_set = set()
+        for metric in v:
+            for sub_type in metric.sub_types:
+                sub_type_names.add(joint_metric_key(metric.name, sub_type.name))
+                if sub_type.priority != -1:
+                    sub_type_with_rank.add(sub_type.name)
+                    rank_set.add(sub_type.priority)
+
+        if not rank_set and len(sub_type_names) == 1:
+            logger.debug(
+                """No priority is specified, but only one sub type
```

Comment:
will the multiple line string also be prepend with the leading whitespace in the multiple line string here?

Reply:
Yes, updated.